Fix integration tests on HPCM stack

bgeltz opened the pull request:
- Relates to #3129.

bgeltz committed:
Account for MPI startup time when making test assertions

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add PBS env variable to batch checker

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Move LD_PRELOAD setting after env setup

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add get_alloc_nodes for PBS

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Refactor power budgets used in integration tests

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Detect the launcher type in test_profile_policy.py

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add sleep 5 to epoch_inference test

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Skip batch-only signals in test

- Also update signal help text to make clear
  they are batch-only.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

lhlawson reviewed (commented):
LEVELZERO signal exclusions and helper text updates LGTM.

bgeltz committed:
Add timeout support to the PALS launcher

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Reduce uncore min for bounds checking to 800 MHz

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Use setsid with test_multi_app.sh

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add PALS support to tutorials

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Align languange standards and avx specification in tutorials

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz reviewed (commented):
https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html
https://gcc.gnu.org/onlinedocs/gcc/Standards.html

bgeltz committed:
Fixup launcher help text

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

lhlawson commented on integration/test/test_gpu_activity_agent.py:
```diff
@@ -26,8 +26,8 @@
 @util.skip_unless_gpu()
-@util.skip_unless_workload_exists("apps/parres/Kernels/Cxx11/dgemm-mpi-cublas")
-@util.skip_unless_workload_exists("apps/parres/Kernels/Cxx11/nstream-mpi-cuda")
+@util.skip_unless_workload_exists("apps/parres/Kernels/Cxx11/dgemm-multigpu-onemkl")
```
```suggestion
@util.skip_unless_workload_exists("apps/parres/Kernels/Cxx11/dgemm-onemkl")
```

bgeltz committed:
Use topo class member

- Resolves issues with running CI.
- Remove trailing whitespace.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>
Change-Id: Ic23851582a8fdc09f676ae5e99e2fb6710979ec7

bgeltz committed:
Account for intermittent sampling of MPI_Init_thread

- When the Controller is able to sample, this time
  must be removed from what is expected in the
  assertion.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Add barriers to test_hint_time.cpp

- Required for precise timing of Epoch hints.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

lhlawson committed:
First pass gpu activity agent test multi-vendor support

Signed-off-by: lhlawson <lowren.h.lawson@intel.com>
Co-authored-by: Brad Geltz <brgeltz@gmail.com>

bgeltz committed:
Do not prune fields that report NaN from reports

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

bgeltz committed:
Fixup test_multi_app.sh

- Increase test timeout.
- Increase dgemm duration to ensure it is sampled.

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

lhlawson committed:
Quick io.py no epoch region fix

Signed-off-by: lhlawson <lowren.h.lawson@intel.com>

cmcantalupo commented on scripts/geopmpy/io.py:
```diff
@@ -877,7 +877,10 @@ def load_reports(self, reports, dir_name, dir_cache, verbose, do_cache):
-                self._epoch_reports_df = pandas.read_hdf(self._report_h5_name, 'epoch_report')
+                try:
+                    self._epoch_reports_df = pandas.read_hdf(self._report_h5_name, 'epoch_report')
+                except:
+                    pass
```
Can we be more specific about what exception is being handled here, and maybe print a warning?

bgeltz replied:
Issue opened to deal with this everywhere: https://github.com/geopm/geopm/issues/3193

cmcantalupo commented on scripts/geopmpy/io.py:
```diff
@@ -1026,13 +1029,16 @@ def _try_float(val):
+                        _add_column('epoch', cc)
+                    epoch_row.update(epoch_data)
+                    epoch_df_list.append(pandas.DataFrame(epoch_row, index=[0]))
+                except:
+                    pass
 
```
Similar issue as on line 882

cmcantalupo commented on scripts/geopmpy/io.py:
```diff
@@ -1057,9 +1063,12 @@ def _try_float(val):
+            epoch_df = pandas.concat(epoch_df_list, ignore_index=True)
+            epoch_df = epoch_df.reindex(columns=self._columns_order['epoch'])
+            self._epoch_reports_df = epoch_df
+        except:
+            pass
```
Again...

cmcantalupo commented on scripts/geopmpy/launcher.py:
```diff
@@ -918,25 +920,15 @@ def get_idle_nodes(self):
         Returns a list of the names of compute nodes that are currently
         available to run jobs.
         """
-        if self._is_slurm_enabled:
```
i'm not sure i understand this change.  I think the original idea is that you may have Slurm as a *resource manager* but use, say, mpiexec from Intel MPI to do *application launch*.  In this way it does make sense to have logic about Slurm in the base class.

cmcantalupo reviewed (commented):
Some questions about parts of the code where all exceptions are quietly suppressed.  Also not  sure about how the Launcher base class should abstract the resource manager as opposed to the MPI application launcher.

bgeltz committed:
Fixups for test_cpu_characterization and test_gpu_activity

Signed-off-by: Brad Geltz <brad.geltz@intel.com>

cmcantalupo reviewed (dismissed):
LGTM

bgeltz committed:
Revert "Moved GPU-CA agent out of beta."

This reverts commit 2a64cbb33c03308e0194d9779871c407c965cf42.
This reverts commit b6e6ea6d83088c5edd943ec4641f3e585a5b287f.
